File: bench/speed/loop.c
/* See README.md. The size comes from argv so nothing folds. */
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    int64_t n = argc > 1 ? atoll(argv[1]) : 300000000;
    int64_t total = 0;
    for (int64_t i = 0; i < n; i++) {
        total += i;
    }
    printf("%" PRId64 "\n", total);
}

File: bench/speed/loop.rs
// See README.md. The size comes from argv so nothing folds.
fn main() {
    let n: i64 = std::env::args().nth(1).map_or(300000000, |a| a.parse().unwrap());
    let mut total: i64 = 0;
    let mut i: i64 = 0;
    while i < n {
        total += i;
        i += 1;
    }
    println!("{total}");
}
